lib/ohloh_scm/adapters/git_svn/pull.rb

```diff
@@ -0,0 +1,69 @@
+module OhlohScm::Adapters
+  class GitSvnAdapter < AbstractAdapter
+    def pull(source_scm, &block)
+      case from
+      when SvnAdapter
+        @source_scm = source_scm
+        convertToGit(&block)
+      else
+        logger.error { "Cannot convert #{from.english_name}/#{from.class_name} repository to git" }
+      end
+    end
+
+    def branch_name
+      'master'
+    end
+
+    private
+
+    def convertToGit
+      yield(1, 2) if block_given?
+
+      if FileTest.exist?(git_path)
+        fetch
+      else
+        clone
+      end
+
+      yield(2, 2) if block_given?
+    end
+
+    def clone
+      # git svn does not support non iteractive and serv-certificate options
+      # Permenently accept svn certificate when it prompts
+      prepare_dest_dir
+      accept_certificate_if_prompted
+      run "#{password_prompt} git svn clone #{username_opts} '#{@source_scm.url}' '#{self.url}'"
+    end
+
+    def accept_certificate_if_prompted
+      run "echo p | svn info #{username_opts} #{password_opts} #{ @source_scm.url }"
+    end
+
+    def password_prompt
+      "echo #{ @source_scm.password } |" unless @source_scm.password.to_s.empty?
+    end
+
+    def password_opts
+      "--password='#{@source_scm.password}'" unless @source_scm.password.to_s.empty?
+    end
+
+    def username_opts
+      "--username #{ @source_scm.username }" unless @source_scm.username.to_s.empty?
+    end
```

Comment:
My concern with `password_prompt`, `password_opts`, and `username_opts` is that they return two different types of values, which can cause problems.  If the reference field is *not* empty, they return a string, but if it is, then they return null. That potentially puts the responsibility on the owner to ensure that they are handling the return from these calls correctly.

Would it make sense for these to do something like 
```ruby
def password_prompt
  @source_scm.password.to_s.empty? ? '' : "echo #{ @source_scm.password } |"
end
```

Reply:
It makes sense.